DM-21454: Include all column names in foreign key names.

TallJimbo opened the pull request:
This is the easiest way to ensure unique foreign key names.

TallJimbo committed:
Include all column names in foreign key names.

This is the easiest way to ensure unique foreign key names.

andy-slac commented on python/lsst/daf/butler/core/schema.py:
```diff
@@ -305,9 +305,10 @@ def toSqlAlchemy(self, tableName: str, schema: Schema) -> ForeignKeyConstraint:
         constraint : `sqlalchemy.ForeignKeyConstraint`
             SQLAlchemy version of the foreign key constraint.
         """
+        name = "_".join(["fkey", tableName, self.table] + list(self.target) + list(self.source))
```
I think this has a potential to exceed name size limits for some backends.

TallJimbo replied:
I'm a bit worried about that, too, but @MichelleGower hasn't reported any problems in her tests yet, and I don't think we can guarantee a unique name without including all of these terms, even if they aren't usually necessary.  Given that this isn't really a useful-visible name (except maybe in some pretty low-level debugging contexts), I think the next step is probably trying to hash these terms into something with fixed size, and I'd rather not do that on this ticket.

TallJimbo committed:
Work around bad SQLAlchemy vs. Oracle assumptions.

TallJimbo committed:
Fix bugs in schema-from-config logic.

MichelleGower committed:
Increase column length for abstract filter

There is an abstract filter called "UNRECOGNISED" in table
PHYSICAL_FILTER which was longer than the column width.
Increased the column width from 8 to 16.

MichelleGower commented:
With these changes and a newer version of SQLAlchemy, I was able to run ci_hsc_gen3 with Oracle.
